File: geometry/pythagorean_theorem/golang/pythagorean.go
package main

import (
	"fmt"
	"math"
)

// findHypotenuse finds the hypotenuse given two sides,
// a and b, of a right triange.
func findHypotenuse(a float64, b float64) float64 {
	return math.Sqrt(a*a + b*b)
}

func main() {
	fmt.Println(findHypotenuse(3, 4))
}

File: geometry/pythagorean_theorem/javascript/pythagorean.js
module.exports = function(a,b) {
  var out_a = a * a;
  var out_b = b * b;
  var result = Math.sqrt(out_a + out_b);
  return {given:[a,b], s2:{ a: out_a, b: out_b }, result}
}